nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -636,168 +514,40 @@ void nano::bootstrap_server::timeout ()
-		}
-
-		if (message_a.query)
-		{
-			boost::optional<std::pair<nano::account, nano::signature>> response (std::make_pair (connection->node->node_id.pub, nano::sign_message (connection->node->node_id.prv, connection->node->node_id.pub, *message_a.query)));
-			debug_assert (!nano::validate_message (response->first, *message_a.query, response->second));
-			auto cookie (connection->node->network.syn_cookies.assign (nano::transport::map_tcp_to_endpoint (connection->remote_endpoint)));
-			nano::node_id_handshake response_message (connection->node->network_params.network, cookie, response);
-			auto shared_const_buffer = response_message.to_shared_const_buffer ();
-			connection->socket->async_write (shared_const_buffer, [connection = std::weak_ptr<nano::bootstrap_server> (connection)] (boost::system::error_code const & ec, std::size_t size_a) {
-				if (auto connection_l = connection.lock ())
-				{
-					if (ec)
-					{
-						if (connection_l->node->config.logging.network_node_id_handshake_logging ())
-						{
-							connection_l->node->logger.try_log (boost::str (boost::format ("Error sending node_id_handshake to %1%: %2%") % connection_l->remote_endpoint % ec.message ()));
-						}
-						// Stop invalid handshake
-						connection_l->stop ();
-					}
-					else
-					{
-						connection_l->node->stats.inc (nano::stat::type::message, nano::stat::detail::node_id_handshake, nano::stat::dir::out);
-						connection_l->finish_request ();
-					}
-				}
-			});
-		}
-		else if (message_a.response)
-		{
-			nano::account const & node_id (message_a.response->first);
-			if (!connection->node->network.syn_cookies.validate (nano::transport::map_tcp_to_endpoint (connection->remote_endpoint), node_id, message_a.response->second) && node_id != connection->node->node_id.pub)
-			{
-				connection->remote_node_id = node_id;
-				connection->socket->type_set (nano::socket::type_t::realtime);
-				++connection->node->bootstrap.realtime_count;
-				connection->finish_request_async ();
-			}
-			else
-			{
-				// Stop invalid handshake
-				connection->stop ();
-			}
-		}
-		else
-		{
-			connection->finish_request_async ();
-		}
-		nano::account node_id (connection->remote_node_id);
-		nano::socket::type_t type = connection->socket->type ();
-		debug_assert (node_id.is_zero () || type == nano::socket::type_t::realtime);
-		connection->node->network.tcp_message_manager.put_message (nano::tcp_message_item{ std::make_shared<nano::node_id_handshake> (message_a), connection->remote_endpoint, connection->remote_node_id, connection->socket });
+		++node->bootstrap.bootstrap_count;
+		socket->type_set (nano::socket::type_t::bootstrap);
+		return true;
 	}
-	std::shared_ptr<nano::bootstrap_server> connection;
-};
+	return false;
 }
 
-void nano::bootstrap_server::run_next (nano::unique_lock<nano::mutex> & lock_a)
+bool nano::bootstrap_server::to_realtime_connection (nano::account const & node_id)
```

Comment:
To match other code we an error code is signalled by returning true, success == 0 == false.

Reply:
I found that convention very misleading, seems like a low level C legacy. Let's say we have a check for success using the false for indicating that: 'if !operation_successful())' that reads "if not operation successful" which is logically the opposite of what we want. Needless cognitive overhead in my opinion.